sonar-plugin-api/src/main/java/org/sonar/api/batch/sensor/highlighting/NewHighlighting.java

```diff
@@ -40,7 +40,9 @@
    * @param startOffset Starting position in file for this type of text. Beginning of a file starts with offset '0'.
    * @param endOffset End position in file for this type of text.
    * @param typeOfText see {@link TypeOfText} values.
+   * @deprecated since 5.6 Only supported to ease migration from old API. Please prefer {@link #highlight(int, int, int, int)}.
    */
+  @Deprecated
   NewHighlighting highlight(int startOffset, int endOffset, TypeOfText typeOfText);
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)